### PR TITLE
prevent rendering if readme is changed

### PR DESCRIPTION
**Pull request by stushurik**
The solution is based on active file name.
Theoretically, it is possible that user might change file during update, so re-run will be completed.
But I assumed that the chance of this event is low, and it is non-obvious user flow.
To avoid that case, it seems that, VizRunner component should know and keep track any change that looks as much more complicated solution.

### CODE REVIEW

File: packages/neoFrontend/src/pages/VizPage/VizRunnerContext/index.js [hunk 1]
```diff
@@ -83,12 +86,19 @@ export const VizRunnerProvider = ({ children }) => {
       if (clearConsole) {
         console.clear();
       }
-      iFrame.setAttribute(
-        'srcDoc',
-        computeSrcDoc(getVizFiles(viz$.getValue()))
-      );
+
+      // render frame if it is initial render or if files that might change viz were changed
```
Comment (curran) [inline packages/neoFrontend/src/pages/VizPage/VizRunnerContext/index.js]: Great solution! Happy that you identified and handled the case of initial render. This is the correct desired behavior.

File: packages/neoFrontend/src/pages/VizPage/VizContext/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { useViz } from './useViz';
 
 export const VizContext = createContext();
 
-export const VizProvider = ({ fallback, children }) => {
```
Comment (curran) [inline packages/neoFrontend/src/pages/VizPage/VizContext/index.js:7]: Good catch!

**Review by curran — CHANGES_REQUESTED** [review]
Generally looks good, but it turns out that it updates whenever the active file is changed, which is not desired. Perhaps a different approach would be better, where `src/pages/VizPage/RunContext/useRun.js` is changed using an approach similar to line 37 there `const jsChanged = useRef(false);`. There could be a new thing maybe called `codeChanged`, and only if the code changed then increment the run id.

**Comment by stushurik** [comment]
@curran i updated PR according to suggested solution

File: packages/neoFrontend/src/pages/VizPage/RunContext/useRun.js [hunk 3]
```diff
@@ -150,7 +159,7 @@ export const useRun = () => {
   // Reset run timer whenever files are changed by the user.
   // Do not reset run timer when bundle.js gets generated.
   useEffect(() => {
-    const subscription = vizContentOp$.subscribe(({ previous, next, op }) => {
+    const subscription = vizContentOp$.subscribe(({ previous, next }) => {
```
Comment (curran) [inline packages/neoFrontend/src/pages/VizPage/RunContext/useRun.js:162]: Surprising that this sort of thing is not caught by the linter part of create-react-app. Maybe we need to iterate on the linting setup (separate from this PR). Thanks for catching this!

File: packages/neoFrontend/src/pages/VizPage/RunContext/changesJS.js [hunk 4]
```diff
@@ -1,28 +1,5 @@
...
-        const extension = getExtension(fileName);
-        if (extension === '.js') {
-          return true;
-        }
-      }
-    }
-  }
-  return false;
-};
+export const changesJS = createChangesChecker('.js');
```
Comment (curran) [inline packages/neoFrontend/src/pages/VizPage/RunContext/changesJS.js:5]: Nice abstraction.

**Review by curran — APPROVED** [review]
Looks great and works well!

**Pull request merged**